src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,67 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationFailureException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_DATA_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+    private StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        userNullDataValidation(user);
+        userLoginAndPasswordValidation(user);
+        ageValidation(user);
+
+        String login = user.getLogin();
+        if (storageDao.get(login) != null) {
+            throw new RegistrationFailureException("This user login exists: " + login);
+        }
+
+        storageDao.add(user);
+        return storageDao.get(login);
```

Comment:
I think `add()` also returns that user, so there is no need to call `get()` after it